### PR TITLE
vm: use same storage account defaults like portal

### LINKED ISSUE #2030: VM: update storage sku defaults
-  default to “Premium_LRS” while creating a VM or Managed Disks.
- “Snapshots” defaulted to “Standard_LRS”. 
- “Images” defaulted to “Standard_LRS”. 
- VMSS still stay at "Standard_LRS"  (portal does that too)

### PR DESCRIPTION
**Pull request by yugangw-msft**
fix #2030

### CODE REVIEW

**Comment by tjprescott** [comment]
Since we are changing defaults, we should annotate this as a breaking change.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def list_ip_addresses(resource_group_name=None, vm_name=None):
 
 
 def create_managed_disk(resource_group_name, disk_name, location=None,
-                        size_gb=None, sku=None,
+                        size_gb=None, sku='Premium_LRS',
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: Is "None" actually a valid size?  Does the server have defaulting logic, or should this be required?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -367,7 +367,7 @@ def attach_managed_data_disk_to_vmss(resource_group_name, vmss_name, size_gb, lu
     data_disks = vmss.virtual_machine_profile.storage_profile.data_disks or []
     if lun is None:
         luns = [d.lun for d in data_disks]
-        lun = max(luns) + 1 if luns else 1
+        lun = max(luns) + 1 if luns else 0
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: Does this have any customer-facing impact besides just 0 indexing LUNs?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: Server default to `Stanard_LRS` if not provided, which is the our old logic.  The change here is to override it

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py (hunk 2 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: No public facing impact. I am getting things consistent with portal behavior

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: Okay, but what about size_gb?  Does None trigger a server-side default or should it actually be a required parameter?

Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: It is only required when you provided no `--source`, we have a validation logic for that.

**Pull request merged**